### ISSUE TITLE
Tumblr embeds cause validation errors and fail to embed

### ISSUE THREAD

**Issue by westonruter** [Automattic/amp-wp#4757]
## Bug Description

When embedding a Tumblr post like https://teded.tumblr.com/post/184736320764/how-do-vaccines-work

I get a validation error for an invalid script:

![image](https://user-images.githubusercontent.com/134745/82635135-676f7e00-9bb4-11ea-9c32-57f2057855d3.png)

And the post fails to embed:

![image](https://user-images.githubusercontent.com/134745/82635200-8bcb5a80-9bb4-11ea-97b5-50a335f57c57.png)

## Expected Behaviour

The AMP version should look the same as the non-AMP verison.

## Steps to reproduce

Add a Tumblr block for https://teded.tumblr.com/post/184736320764/how-do-vaccines-work

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

I think all that is needed is fixing the pattern for the `data-href` attribute:

```diff
- if ( preg_match( '#data-href="(?P<href>https://embed.tumblr.com/embed/post/\w+/\w+)"#', $cache, $matches ) ) {
+ if ( preg_match( '#data-href="(?P<href>https://embed.tumblr.com/embed/post/[^"]+?)"#', $cache, $matches ) ) {
```

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
@pierlon I'm assuming this should be handled as part of #4650.

**Comment by pierlon** [comment]
Yes it is, see 21ae657d0c3228051f9138941da32759c38dd387.

**Comment by westonruter** [comment]
Since #4650 is too large for 2.1 now, either the relevant change can be cherry-picked for 2.1 to fix Tumblr, or else we just punt to 2.2 altogether.

**Comment by jwold** [comment]
**QA Passed**

**Steps I took:**
1. Install AMP on fresh WordPress site using Standard mode
2. Grabbed Tumblr URL and embedded as block
3. Confirmed that AMP checkmark is green, and no errors present on site
4. Scrolled down to see entire Tumblr post listed in the block on the WordPress page

<img width="988" alt="Screen Shot 2021-04-26 at 7 46 30 AM" src="https://user-images.githubusercontent.com/3411173/116102575-ae3ae300-a663-11eb-8ca3-fafb85f05b4a.png">

<img width="1280" alt="Screen Shot 2021-04-26 at 7 48 32 AM" src="https://user-images.githubusercontent.com/3411173/116102778-d4f91980-a663-11eb-8ac3-d546b9274188.png">
